cmd/server/main.go

```diff
@@ -24,23 +25,28 @@ func main() {
 	var s = server.GetStorage(handlers.Restore, handlers.DatabaseDSN)
 
 	if handlers.DatabaseDSN != "" {
-		dml.DBRepository = *dml.NewRepository()
-		err := db.Init(&dml.DBRepository)
+		rep, err := dml.NewRepository()
+		if err != nil {
+			logger.WriteErrorLog(err.Error(), "NewRepository")
+			return
+		}
+		dml.DBRepository = *rep
+		err = db.Init(&dml.DBRepository)
 		defer dml.DBRepository.Close()
 		if err != nil {
 			logger.WriteErrorLog("Error in init db", err.Error())
+			return
 		}
 	}
 
-	ticker := time.NewTicker(time.Duration(handlers.StoreInterval) * time.Second)
-	go func() {
-		if handlers.StoreInterval == 0 {
-			return
-		}
-		server.SaveMetricsPerTime(server.MaxSaverWorkTime, ticker, s)
-	}()
+	if handlers.Restore && handlers.StoreInterval > 0 {
```

Comment:
```suggestion
	if handlers.StoreInterval > 0 && writingToFileIsEnabledAndAvailable {
```
Периодическое сохранение никак не связано с функционалом подгрузки содержимого файла в память при запуске. Оно зависит только от флагов `-f` и `-i` (ну и соответствующих переменных окружения).